src/rust/engine/graph/src/entry.rs

```diff
@@ -428,6 +471,9 @@ impl<N: Node> Entry<N> {
             "Not completing node {:?} because it was invalidated before completing.",
             self.node
           );
+          if let Some(previous_result) = previous_result.as_mut() {
```

Comment:
Nit: Possible that the `EntryResult` enum could gain a "NotPresent" variant to incorporate the `None` case? Possibly not worth it.

Reply:
Will leave for now, can add in the future if needed.